### ISSUE TITLE
MyEnum.V style typing does not work with pytest

### ISSUE THREAD

**Issue by tdudz** [dropbox/mypy-protobuf#295]
while it works and passes everything with mypy 0.910, when i run my pytest on my codebase it says:

`AttributeError: Enum MyEnum has no value defined for name 'V'`

is the only way to fix this my going through my entire codebase and replacing `MyEnum.V` with a version wrapped in quotes, like `"MyEnum.V"` ?

**Comment by nipunn1313** [comment]
Hiya - can you provide a code example?

**Comment by nipunn1313** [comment]
There's some documentation on some nuances of the `V` here
https://github.com/dropbox/mypy-protobuf#types-enum-int-values-more-strongly

**Comment by nipunn1313** [comment]
It's likely you need
```
from __future__ import annotations  # Not needed with python>=3.10
```

The `.V` doesn't exist at runtime until #169 upstreams into protobuf itself.

**Comment by tdudz** [comment]
interestingly enough, i already have that imported (i just upgraded from `mypy-protobuf` v1.23 to v2.5)

the code is nothing special, happens in a bunch of places wherever i define the type. one particular case is when i alias a new type:

`Foo = Tuple[MyEnum.V, MyEnum2.V, MyEnum3.V]`

**Comment by nipunn1313** [comment]
Yep, unfortunately, `.V` doesn't exist at runtime (yet).

You have a few options when dealing with type aliases involving enums
- Use quotes
- Hide them behind `if typing.TYPE_CHECKING` so it doesn't execute at runtime [docs](https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING)


I can add docs for your particular use cases in README to help the future traveller

**Comment by tdudz** [comment]
according to this it should work, im confused why importing annotations didnt fix it. is it just when aliasing?

<img width="808" alt="Screen Shot 2021-09-17 at 7 11 11 PM" src="https://user-images.githubusercontent.com/15475123/133862587-82d5cbbc-e167-4679-b25f-54ed5f1c5517.png">

**Comment by nipunn1313** [comment]
Updated docs with https://github.com/dropbox/mypy-protobuf/commit/43ae0fb534c2e930b92c6e796014fec05be3928d

To answer your question, aliasing is not considered an annotation.

```
# This is an annotation. Python >= 3.10 does not actually evaluate the annotation at runtime per PEP563
a: MyEnum.V = 5

# This is not an annotation. Python cannot determine that it's a type alias at runtime until it's executed.
FOO = Tuple[MyEnum.V, MyEnum.V]

# This is also not an annotation - python runtime interprets it as a function call with two args
cast(MyEnum.V, x)
```

**Comment by tdudz** [comment]
got it -- thanks! will likely resort to strings for the time being

unrelated but just as a note, i use conda and conda hasn't published protobuf 3.18.0 yet, so i'm unable to use latest mypy-protobuf. looks like 2.5 is latest compatible for my env
